app/src/main/java/cloudcity/util/CloudCityLogger.java

```diff
@@ -0,0 +1,67 @@
+package cloudcity.util;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import okhttp3.logging.HttpLoggingInterceptor;
+import timber.log.Timber;
+
+public class CloudCityLogger {
+
+    public static void d(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).d(th, message);
+    }
+
+    public static void d(String logtag, String message) {
+        d(logtag, message, null);
+    }
+
+    public static void v(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).v(th, message);
+    }
+
+    public static void v(String logtag, String message) {
+        v(logtag, message, null);
+    }
+
+    public static void i(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).i(th, message);
+    }
+
+    public static void i(String logtag, String message) {
+        i(logtag, message, null);
+    }
+
+    public static void w(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).w(th, message);
+    }
+
+    public static void w(String logtag, String message) {
+        w(logtag, message, null);
+    }
+
+    public static void e(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).e(th, message);
+    }
+
+    public static void e(String logtag, String message) {
+        e(logtag, message, null);
+    }
+
+    public static void wtf(String logtag, String message, Throwable th) {
+        Timber.tag(logtag).wtf(th, message);
+    }
+
+    public static void wtf(String logtag, String message) {
+        wtf(logtag, message, null);
+    }
```

Comment:
as I've said multiple times, rather than suggesting unnecessarily overly-defensive programming, look into the receiver's code and see if it isn't handling nulls properly before suggesting that.

Case being, Timber won't do anything if the message is null and will just silently swallow it. In fact, it already does everything you're suggesting us to do here. resolving.